go/vt/vtctl/grpcvtctldserver/query.go

```diff
@@ -0,0 +1,312 @@
+		return nil, err
+	}
+
+	sm.CleanedUpAt, err = valueToVTTime(row[15])
+	if err != nil {
+		return nil, err
+	}
+
+	if status := row[16].ToString(); status != "" {
+		sm.Status, err = schematools.ParseSchemaMigrationStatus(status)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	sm.LogPath = row[17].ToString()
+	sm.Artifacts = row[18].ToString()
+
+	sm.Retries, err = row[19].ToUint64()
+	if err != nil {
+		return nil, err
+	}
+
+	if alias := row[20].ToString(); alias != "" {
+		sm.Tablet, err = topoproto.ParseTabletAlias(alias)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	sm.TabletFailure, err = row[21].ToBool()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.Progress, err = row[22].ToFloat32()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.MigrationContext = row[23].ToString()
+	sm.DdlAction = row[24].ToString()
+	sm.Message = row[25].ToString()
+
+	sm.EtaSeconds, err = row[26].ToInt64()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.RowsCopied, err = row[27].ToUint64()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.TableRows, err = row[28].ToInt64()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.AddedUniqueKeys, err = row[29].ToUint32()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.RemovedUniqueKeys, err = row[30].ToUint32()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.LogFile = row[31].ToString()
+
+	sm.ArtifactRetention, err = valueToVTDuration(row[32], "s")
+	if err != nil {
+		return nil, err
+	}
+
+	sm.PostponeCompletion, err = row[33].ToBool()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.RemovedUniqueKeyNames = row[34].ToString()
+	sm.DroppedNoDefaultColumnNames = row[35].ToString()
+	sm.ExpandedColumnNames = row[36].ToString()
+	sm.RevertibleNotes = row[37].ToString()
+
+	sm.AllowConcurrent, err = row[38].ToBool()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.RevertedUuid = row[39].ToString()
+
+	sm.IsView, err = row[40].ToBool()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.ReadyToComplete, err = row[41].ToBool()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.VitessLivenessIndicator, err = row[42].ToInt64()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.UserThrottleRatio, err = row[43].ToFloat32()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.SpecialPlan = row[44].ToString()
+
+	sm.LastThrottledAt, err = valueToVTTime(row[45])
+	if err != nil {
+		return nil, err
+	}
+
+	sm.ComponentThrottled = row[46].ToString()
+
+	sm.CancelledAt, err = valueToVTTime(row[47])
+	if err != nil {
+		return nil, err
+	}
+
+	sm.PostponeLaunch, err = row[48].ToBool()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.Stage = row[49].ToString()
+
+	sm.CutoverAttempts, err = row[50].ToUint32()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.IsImmediateOperation, err = row[51].ToBool()
+	if err != nil {
+		return nil, err
+	}
+
+	sm.ReviewedAt, err = valueToVTTime(row[52])
+	if err != nil {
+		return nil, err
+	}
+
+	sm.ReadyToCompleteAt, err = valueToVTTime(row[53])
+	if err != nil {
+		return nil, err
+	}
+
+	return sm, nil
```

Comment:
Let's switch away from `row[42].ToInt64()` to `row.AsInt64("column_name", <defaultvalue>)`. To do that, use `rows.Named()` which give you a slice of decorated row objects. See for example:

https://github.com/vitessio/vitess/blob/30d90fbc76c317286aaafa2f9180946cc124dbd0/go/vt/vttablet/onlineddl/executor.go#L3496-L3503

The reason we should make the columns named rather than use their indexes, is that nothing forces me to keep table columns at their current order. I may, and will, add new columns in the middle, or rearrange their order. So we can't rely on the column positioning.

All that assuming, of course, you're running a `SELECT *`. But if you `SELECT this_column, that_column` and want to rely on that order, feel free to ignore the comment.

Reply:
no this is way better! i'm new to `Named()` this is awesome